core/services/relay/evm/mercury/v1/data_source_test.go

```diff
@@ -202,25 +195,21 @@ func TestMercury_Observe(t *testing.T) {
 					assert.NoError(t, obs.MaxFinalizedBlockNumber.Err)
 					assert.Equal(t, head.Number-1, obs.MaxFinalizedBlockNumber.Val)
 				})
-				t.Run("if current block num errored", func(t *testing.T) {
+				t.Run("if no current block available", func(t *testing.T) {
 					h2 := commonmocks.NewHeadTracker[*evmtypes.Head, common.Hash](t)
 					h2.On("LatestChain").Return((*evmtypes.Head)(nil))
 					ht.h = h2
-					c2 := evmclimocks.NewClient(t)
-					c2.On("HeadByNumber", mock.Anything, (*big.Int)(nil)).Return(nil, errors.New("head retrieval failed"))
-					ht.c = c2
 
 					obs, err := ds.Observe(ctx, repts, true)
 					assert.NoError(t, err)
 
-					assert.EqualError(t, obs.MaxFinalizedBlockNumber.Err, "FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed. No initialBlockNumber was set, tried to use current block number to determine maxFinalizedBlockNumber but got error: head retrieval failed")
+					assert.EqualError(t, obs.MaxFinalizedBlockNumber.Err, "FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed. No initialBlockNumber was set, tried to use current block number to determine maxFinalizedBlockNumber but got error: no blocks available")
```

Comment:
This seems fit for a custom error:
```go
type ErrWhatever type {
	Err error
}

func () UnWrap() error { return e.Err}

func () Error() string{
	return fmt.Sprintf("FetchInitialMaxFinalizedBlockNumber returned empty LatestReport; this is a new feed. No initialBlockNumber was set, tried to use current block number to determine maxFinalizedBlockNumber but got error: %v", e.Err)
}
```
Then here you can just do:
```suggestion
					assert.ErrorIs(t, obs.MaxFinalizedBlockNumber.Err, ErrWhatever{})
```